lib/html.js

```diff
@@ -490,15 +490,31 @@ var appendDOM = function(elem, child) {
       }
 
       // in IE must explicitly nest TRs in TBODY
-      var childTag = child.tagName.toLowerCase();// child tagName
+      var childTag = child.tagName.toLowerCase();
       if (childTag && childTag !== 'tbody' && childTag !== 'thead') {
-        // insert in last tbody
-        var tBody = elem.tBodies.length > 0 ? elem.tBodies[elem.tBodies.length-1] : null;
-        if (!tBody) {
-          tBody = createElement(childTag === 'th' ? 'thead' : 'tbody');
-          elem.appendChild(tBody);
+        var isHeader = childTag === 'th';
+        // insert in existing tHead or last tBody
+        var section = isHeader ? elem.tHead : elem.tBodies[elem.tBodies.length - 1];
+        if (!section) {
+          section = createElement(isHeader ? 'thead' : 'tbody');
+          elem.appendChild(section);
+        }
+        if (childTag === 'tr') {
+          section.appendChild(child);
+        } else {
+          // find last tr
+          var tr;
+          for (var i = section.childNodes.length - 1; i >= 0 && !tr; i--) {
```

Comment:
The comment says find the last one, but wouldn't you want to break out of the loop once you have a value for `tr` here since it's iterating backwards? Or am I missing something?

Reply:
`i >= 0 && !tr`

I could explicitly `break` if that's easier to read. (probably)

Reply:
I am missing something! `&& !tr` is sneaky in there

Reply:
i'll change this to be less sneaky :)

Reply:
Heh, saw it right as you commented... yeah, I think it'd be a bit easier to read if it were more explicit. My mind scans over the line entirely when I see a `for` and just assumes it's straight iteration over elements but that could just be me 🤷‍♂